.github/actions/check-image-and-changes/action.yaml

```diff
@@ -26,103 +26,173 @@ runs:
       id: changes
       shell: bash
       run: |
-        # Ensure that SHORT_SHA is available for subsequent steps in this context
-        echo "SHORT_SHA=${SHORT_SHA:-${{ steps.get-sha.outputs.short_sha }}}" >> $GITHUB_ENV
-
-        # Get base commit for comparison
-        BASE_COMMIT=$(git merge-base ${{ github.event.pull_request.base.sha }} HEAD)
-        echo "BASE_COMMIT=$BASE_COMMIT"
-
-        # Get list of changed files
-        CHANGED_FILES=$(git diff --name-only "$BASE_COMMIT" HEAD)
-        echo "Changed files:"
-        echo "$CHANGED_FILES"
-
-        # Check for [skip-build] tag in commit messages
-        SKIP_BUILD_TAG=false
-        COMMIT_MESSAGES=$(git log --format=%B "$BASE_COMMIT"..HEAD)
-        if echo "$COMMIT_MESSAGES" | grep -q "\[skip-build\]"; then
-          echo "Found [skip-build] tag in commit messages."
-          SKIP_BUILD_TAG=true
+        echo "🔍 Checking changes and build requirements..."
+        
+        # Function to check if file is in always-skip paths
+        check_always_skip_paths() {
+          local file=$1
+          if [[ "$file" == docs/* ]] || [[ "$file" == *.md ]]; then
+            return 0  # true - always skip
+          fi
+          return 1  # false - don't skip automatically
+        }
+        
+        # Function to check if file is in image-dependent paths
+        check_image_dependent_paths() {
+          local file=$1
+          if [[ "$file" == e2e/* ]] || [[ "$file" == e2e-tests/* ]] || [[ "$file" == .ibm/* ]] || [[ "$file" == .rhdh/docker/* ]]; then
```

Comment:
@nickboldt Good point about the RPM lock files. We should definitely keep this in mind for future compatibility. 

When those files are moved upstream, we'll need to update the `check_image_dependent_paths()` function to include the new paths. I'll add a TODO comment or we can create an issue to track this dependency.

Thanks for the heads up! 🙏